src/main/java/frc/robot/subsystems/swerve/SwerveSubsystem.java

```diff
@@ -363,6 +376,62 @@ public Pose2d getVirtualTarget() {
             .inverse());
   }
 
+  /**
+   * Gets the pose at some time in the future
+   *
+   * @param time time in seconds
+   * @return The future pose
+   */
+  public Pose2d getFuturePose(double time) {
+    return getPose()
+        .transformBy(
+            new Transform2d(
+                getRobotRelativeSpeeds().vxMetersPerSecond * time,
+                getRobotRelativeSpeeds().vyMetersPerSecond * time,
+                new Rotation2d()));
+  }
+
+  @AutoLogOutput(key = "AutoAim/FuturePose")
+  public Pose2d getFuturePose() {
```

Comment:
make sure the doc comment appears on this one too, and if it doesnt copy it and add that it defaults to using a fixed lookahead time